blog/_posts/2024-06-06-istio-native-sidecar.md

```diff
@@ -0,0 +1,164 @@
+---
+title: "Introducing Native Sidecar Support for Istio addon on AKS"
+description: "Explore native sidecar support for the Istio add-on in Azure Kubernetes Service (AKS). Learn its benefits, how to enable it, and see a practical demo."
+date: 2024-06-06
+author: Fuyuan Bie
+categories:
+  - networking
+  - addons
+---
+
+We're excited to announce the public preview of native sidecar support for Istio add-on for Azure Kubernetes Service (AKS). This blog post will guide you through the fundamentals of Kubernetes native sidecars, the integration of native sidecar support in Istio add-on, the benefits of this feature, how to enable it, and a demo showcasing its advantages.
+
+## Understanding Kubernetes Native Sidecars
+
+In Kubernetes, the sidecar pattern is used to extend the functionality of applications running in pods. Sidecar containers are responsible for auxiliary tasks such as logging, proxying, and monitoring. Typically, managing sidecar lifecycle required special configurations, which could be complex and error-prone. Developers must understand the lifecycle interactions of containers within a Kubernetes pod to avoid unexpected behaviors. They cannot safely assume that sidecars will be ready before their application or expect a specific shutdown order. Additionally, for jobs, the primary job container must terminate the sidecars.
+
+Starting from Kubernetes v1.28, native sidecar support becomes a builtin feature. On AKS, starting from v1.29, this feature is turned on.
```

Comment:
let's remove the k8s 1.28 part, it was in alpha in 1.28. for AKS we start consuming features as it gets to beta. we can instead mention that native sidecars reached beta in k8s upstream 1.29 and that AKS as a result has it too starting 1.29 now